File: apps/trader-cli/src/symbols.ts
/** 用户偏好研究标的（UI 快选 / 批量 scan 默认范围，非系统硬限制） */
export const PREFERRED_SYMBOLS = [
  "TSLA",
  "TSLL",
  "QQQ",
  "SPY",
  "ARKK",
  "NVDA",
  "COIN",
  "BMNR",
] as const;

/** @deprecated 别名，逐步改用 PREFERRED_SYMBOLS */
export const MVP_SYMBOLS: readonly string[] = PREFERRED_SYMBOLS;

export const PREFERRED_SYMBOLS_LABEL = PREFERRED_SYMBOLS.join("、");

File: apps/trader-chart/src/symbols.rs
pub const PREFERRED: &[&str] = &[
    "TSLA", "TSLL", "QQQ", "SPY", "ARKK", "NVDA", "COIN", "BMNR",
];

pub fn normalize_ticker(raw: &str) -> Option<String> {
    let t = raw.trim().to_uppercase();
    if t.is_empty() || !t.chars().all(|c| c.is_ascii_alphanumeric() || c == '.' || c == '-') {
        return None;
    }
    Some(t)
}

pub fn cycle_symbol(current: &str, delta: i32) -> String {
    let cur = normalize_ticker(current).unwrap_or_else(|| PREFERRED[0].to_string());
    let idx = PREFERRED.iter().position(|&s| s == cur).unwrap_or(0);
    let len = PREFERRED.len() as i32;
    let next = (idx as i32 + delta).rem_euclid(len) as usize;
    PREFERRED[next].to_string()
}
